tests/transforms/test_unitary_to_rot.py

```diff
@@ -350,3 +357,246 @@ def qfunc_with_qubit_unitary(angles):
+
+        def two_qubit_decomp_qnode(x):
+            qml.RX(x, wires=0)
+            qml.QubitUnitary(U0, wires=[0, 1])
+            qml.QubitUnitary(U1, wires=[1, 2])
+            return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1) @ qml.PauliZ(2))
+
+        x = tf.Variable(0.1, dtype=tf.float64)
+
+        transformed_x = tf.Variable(0.1, dtype=tf.float64)
+
+        dev = qml.device("default.qubit", wires=3)
+
+        original_qnode = qml.QNode(
+            two_qubit_decomp_qnode, dev, interface="tf", diff_method=diff_method
+        )
+
+        original_result = original_qnode(x)
+
+        transformed_qfunc = unitary_to_rot(two_qubit_decomp_qnode)
+        transformed_qnode = qml.QNode(
+            transformed_qfunc, dev, interface="tf", diff_method=diff_method
+        )
+
+        transformed_result = transformed_qnode(transformed_x)
+
+        assert qml.math.allclose(original_result, transformed_result)
+
+        assert len(transformed_qnode.qtape.operations) == 13
+
+        with tf.GradientTape() as tape:
+            loss = original_qnode(x)
+        original_grad = tape.gradient(loss, x)
+
+        with tf.GradientTape() as tape:
+            loss = transformed_qnode(transformed_x)
+        transformed_grad = tape.gradient(loss, transformed_x)
+
+        # For 64bit values, need to slightly increase the tolerance threshold
+        assert qml.math.allclose(original_grad, transformed_grad, atol=1e-7)
+
+    @pytest.mark.parametrize("diff_method", ["parameter-shift", "backprop"])
+    def test_gradient_unitary_to_rot_two_qubit_jax(self, diff_method):
+        """Tests differentiability in jax interface."""
+        jax = pytest.importorskip("jax")
+        from jax import numpy as jnp
+
+        from jax.config import config
+
+        remember = config.read("jax_enable_x64")
+        config.update("jax_enable_x64", True)
+
+        U0 = jnp.array(test_two_qubit_unitaries[0], dtype=jnp.complex128)
+        U1 = jnp.array(test_two_qubit_unitaries[1], dtype=jnp.complex128)
+
+        def two_qubit_decomp_qnode(x):
+            qml.RX(x, wires=0)
+            qml.QubitUnitary(U0, wires=[0, 1])
+            qml.QubitUnitary(U1, wires=[1, 2])
+            return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1) @ qml.PauliZ(2))
+
+        x = jnp.array(0.1, dtype=jnp.float64)
+
+        dev = qml.device("default.qubit", wires=3)
+
+        original_qnode = qml.QNode(
+            two_qubit_decomp_qnode, device=dev, interface="jax", diff_method=diff_method
+        )
+
+        transformed_qfunc = unitary_to_rot(two_qubit_decomp_qnode)
+        transformed_qnode = qml.QNode(
+            transformed_qfunc, dev, interface="jax", diff_method=diff_method
+        )
+
+        assert qml.math.allclose(original_qnode(x), transformed_qnode(x))
+
+        # 3 normal operations + 10 for the first decomp and 2 for the second
+        assert len(transformed_qnode.qtape.operations) == 13
+
+        original_grad = jax.grad(original_qnode, argnums=(0))(x)
```

Comment:
I was having issues with this test locally; it's like it was not recognizing that I was specifying to differentiate only to argnum 0. This doesn't seem to be causing an issue in the CI-run tests though.